src/TreeView/index.tsx

```diff
@@ -1053,9 +1048,20 @@ const Node = (props: INodeProps) => {
         lastInteractedWith: element.id,
       });
     } else if (event.ctrlKey || clickActions.select) {
+      const isSelectedAndHasSelectedDescendants =
```

Comment:
Can you move this in utils? You can reuse it then in both places without code duplicaition.

Reply:
fixed